docs/how-it-works.md

```diff
@@ -0,0 +1,60 @@
+# How Turbinia Works
+## General
+This page contains some of the details on the internals of how Turbinia works.
+
+## Tasks
+A Task is the smallest discrete unit of schedulable work.  Tasks are sheduled
+and executed on remote Workers.  A Task can generate more Evidence that is
+returned back to the Turbinia server for possible further processing.
+
+## Jobs
+A Job is a larger unit of work that creates one or more Tasks to process that
+work.  A Job can either create a single Task, or it can break up that work and
+create multiple Tasks.  Sometimes these terms are used interchangeably (though
+we mostly talk about things in terms of Tasks) because in most cases a Job will
+just create a single Task.
+
+## Workers
+Workers are independent processes that run either in Cloud GCE instances or on
+local machines.  They run continuously and wait for new Tasks to be scheduled.
+When a new Task is scheduled, a pseudo-random worker will pick up the Task and
+execute it.  After the Task is complete (successfully or not), the Worker will
+return the status, any error logs and results.
+
+## Task Manager
+The Task Manager Acts as a broker between the clients and workers
+and handles management of Evidence, Jobs, Tasks and Workers
+
+## Task manager flow
+Tasks are configured to "listen" for specific Evidence types, and if the Task
+Maanger sees a new piece of Evidence (either from a new Turbinia request, or
+because another Task generated a new piece of Evidence), and there is a Task
+that configured to run for that type of Evidence, then the Task Maanger will
+schedule a new Task for it.
+
+## Evidence 
```

Comment:
I suggest we put this before the Task manager stuff above seeing we are referencing it there.

Reply:
Done.